lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1898,6 +1898,49 @@ LogicalResult AttachOp::canonicalize(AttachOp op, PatternRewriter &rewriter) {
   return failure();
 }
 
+/// Replaces the given op with the contents of the given single-block region.
+static void replaceOpWithRegion(PatternRewriter &rewriter, Operation *op,
+                                Region &region) {
+  assert(llvm::hasSingleElement(region) && "expected single-region block");
+  Block *fromBlock = &region.front();
+  // Merge it in above the specified operation.
+  op->getBlock()->getOperations().splice(Block::iterator(op),
+                                         fromBlock->getOperations());
+}
+
+LogicalResult WhenOp::canonicalize(WhenOp op, PatternRewriter &rewriter) {
+  if (auto constant = op.getCondition().getDefiningOp<firrtl::ConstantOp>()) {
```

Comment:
Is this new behavior?  If so, please add a test for this folding of whens of constants.  Seems useful, thanks!